wsmaster/che-core-api-devfile/src/main/java/org/eclipse/che/api/devfile/server/DevfileConverter.java

```diff
@@ -125,8 +136,15 @@ public WorkspaceConfigImpl devFileToWorkspaceConfig(Devfile devfile)
           break;
         case KUBERNETES_TOOL_TYPE:
         case OPENSHIFT_TOOL_TYPE:
-          // this kind of tool ignored here since it contain only reference to tool configuration
-          // which should be resolved and provisioned into workspace config recipe separately.
+          try {
+            EnvironmentImpl environment =
+                devfileEnvironmentFactory.createEnvironment(tool, recipeFileContentProvider);
+            final String environmentName = tool.getName();
+            config.setDefaultEnv(environmentName);
+            config.setEnvironments(singletonMap(environmentName, environment));
+          } catch (IllegalArgumentException e) {
+            throw new DevfileFormatException(e.getMessage());
```

Comment:
Hard to say certainly. This will gonna happen if someone provide devfile with recipe tools to the bare devfile API, which has no content supplier. So in this case format error looks more or less logical for me.

Reply:
will add explanation message